eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ChannelOpenSingleFunded.scala

```diff
@@ -377,13 +373,23 @@ trait ChannelOpenSingleFunded extends SingleFundingHandlers with ErrorHandlers {
     case Event(remoteChannelReady: ChannelReady, d: DATA_WAIT_FOR_FUNDING_CONFIRMED) =>
       if (remoteChannelReady.alias_opt.isDefined && d.commitments.localParams.isInitiator) {
         log.info("this channel isn't zero-conf, but we are funder and they sent an early channel_ready with an alias: no need to wait for confirmations")
-        val (shortIds, localChannelReady) = acceptFundingTx(d.commitments, RealScidStatus.Unknown)
-        self ! remoteChannelReady
         // NB: we will receive a WatchFundingConfirmedTriggered later that will simply be ignored
-        goto(WAIT_FOR_CHANNEL_READY) using DATA_WAIT_FOR_CHANNEL_READY(d.commitments, shortIds, localChannelReady) storing() sending localChannelReady
-      } else {
-        log.debug("received their channel_ready, deferring message")
-        stay() using d.copy(deferred = Some(remoteChannelReady)) // no need to store, they will re-send if we get disconnected
+        blockchain ! WatchPublished(self, d.commitments.fundingTxId)
```

Comment:
This is a bit weird: we rely on our peer to tell us whether the tx was published. But they could do what we were doing before and this would be inaccurate.

Reply:
> This is a bit weird: we rely on our peer to tell us whether the tx was published.

I don't understand what you mean, we're not relying on them to tell us if the tx was published, we set a watch to see it published in our own mempool? They're simply signaling that they're ok with zero-conf by sending `channel_ready`, so we agree to do zero-conf once we've made sure the tx was published.

Reply:
I'm wearing 👓 today, but I still misread that you were emitting the _TransactionPublished_ event 🤷‍♂️.